modules/post/multi/gather/enum_software_versions.rb

```diff
@@ -0,0 +1,128 @@
+    when 'linux'
+      # All of the following options were taken from https://distrowatch.com/dwres.php?resource=package-management
+      # and further verified against VMs that were set up in testing labs.
+      if command_exists?('apt') # Debian, Ubuntu, and Debian derived distros.
+        cmd = %w[apt list --installed]
+      elsif command_exists?('dpkg') # Alternative for Debian based systems
+        cmd = %w[dpkg -l]
+      elsif command_exists?('pacman') # Arch and Manjaro are two popular examples
+        cmd = %w[pacman -Q]
+      elsif command_exists?('zypper') # OpenSUSE is a popular example
+        cmd = %w[zypper search -is]
+      elsif command_exists?('rpm') # Fedora, Centos, RHEL
+        cmd = %w[rpm -qa]
+      elsif command_exists?('apk') # Apline
+        cmd = %w[apk info -v]
+      elsif command_exists?('qlist') # Gentoo
+        cmd = %w[qlist -Iv]
+      elsif command_exists?('pkg') # FreeBSD
+        cmd = %w[pkg info]
+      elsif command_exists?('equo') # Sabayon
+        cmd = %w[equo q list installed -v]
+      elsif command_exists?('nix-env')
+        cmd = %w[nix-env -q]
+      else
+        print_error("The target system either doesn't have a package manager system, or does not use a known package manager system!")
+        print_error('Unable to enumerate the softwaare on the target system. Exiting...')
+        return nil
+      end
+
+      if command_exists?((cmd[0]).to_s) == false
+        print_error("The command #{cmd[0]} was not found on the target.")
+        return
+      else
+        listing = cmd_exec(cmd.join(' ')).to_s
+        if listing.empty?
+          print_error('No results were returned when trying to get software installed on the Linux host. An error likely occured.')
+          return
+        end
+        store_linux_loot(listing)
+      end
+    when 'bsd', 'solaris'
+      if command_exists?('pkg') == false
+        print_error("The command 'pkg' does not exist on the host")
+        return
+      end
+      listing = cmd_exec('pkg info').to_s
+      if listing.empty?
+        print_error('No results were returned when trying to get software installed on the BSD/Solaris host. An error likely occured.')
+        return
+      end
+      file = store_loot('host.bsd.solaris.software.versions', 'text/plain', session, listing, 'installed_software.txt', 'Installed Software and Versions')
+      print_good("Stored information about the installed products to the loot file at #{file}")
+    when 'osx'
+      if command_exists?('system_profiler') == false
+        print_error("The command 'system_profiler' does not exist on the host")
+        return
+      end
+      listing = cmd_exec('system_profiler SPApplicationsDataType').to_s
+      if listing.empty?
+        print_error('No results were returned when trying to get software installed on the OSX host. An error likely occured.')
+        return
+      end
```

Comment:
Would checking for `Homebrew` here apply?

Reply:
Not a bad idea to be honest, though I'm not sure if this would affect things or not. Technically speaking, `system_profiler` is Apple's way of profiling everything about the system including the installed software so I'd think it would also pick up what software `Homebrew` installed, but since `Homebrew` is also so popular its a valid point to point out that if `system_profiler` does not pick up on software `Homebrew` installs, that we should also be using `Homebrew` if available to determine what is installed.

Reply:
Okay well looks like the OSX command to do this for `Homebrew` would be `brew list --versions`, according to https://github.com/Homebrew/legacy-homebrew/issues/245. So that should work for that instance. Also I noticed MacPorts is also an option, which we could add in with `port installed` which will grab the versions installed.